Packages/MatchCore/Sources/ProductPurchaseLogic/ProductPurchase.swift

```diff
@@ -44,13 +45,32 @@ public struct ProductPurchaseLogic {
       case .onTask:
         analytics.logScreen(screenName: "ProductPurchase", of: self)
 
-        let ids = build.infoDictionary("PRODUCTS", for: [String].self)!
+        return .run { send in
+          for try await productList in api.productList() {
+            await send(.productListResponse(Result {
+              productList
+            }))
+          }
+        } catch: { error, send in
+          await send(.productListResponse(.failure(error)))
+        }
+        .cancellable(id: Cancel.products, cancelInFlight: true)
```

Comment:
Queryのレスポンスではdataが返ってきて、その中にdata.productListが存在するので、命名としてはdataの方が分かりやすいと思います！
また、現状はdata.productListのみですが、data.currentUserなど他のクエリレスポンスが追加される場合もあります！
```suggestion
          for try await data in api.productList() {
            await send(.productListResponse(Result {
              data
            }))
          }
        } catch: { error, send in
          await send(.productListResponse(.failure(error)))
        }
        .cancellable(id: Cancel.products, cancelInFlight: true)
```